src/controllers/QueryRunner.ts

```diff
@@ -100,70 +104,39 @@ export default class QueryRunner {
 
     // Pulls the query text from the current document/selection and initiates the query
     public runQuery(selection: ISelectionData): Thenable<void> {
-        this._vscodeWrapper.logToOutputChannel(Utils.formatString(Constants.msgStartedExecute, this._uri));
         const self = this;
-        this.batchSets = [];
+        this._vscodeWrapper.logToOutputChannel(Utils.formatString(Constants.msgStartedExecute, this._uri));
+
+        // Put together the request
         let queryDetails: QueryExecuteParams = {
             ownerUri: this._uri,
             querySelection: selection
         };
+
+        // Update internal state to show that we're executing the query
         this._resultLineOffset = selection ? selection.startLine : 0;
         this._isExecuting = true;
         this._statusView.executingQuery(this.uri);
 
+        // Send the request to execute the query
         return this._client.sendRequest(QueryExecuteRequest.type, queryDetails).then(result => {
+            // The query has started, so lets fire up the result pane
             self.eventEmitter.emit('start');
-            if (result.messages) { // Show informational messages if there was no query to execute
-                self._statusView.executedQuery(self.uri);
-                self._isExecuting = false;
-                self.batchSets = [{
-                        hasError: false,
-                        id: 0,
-                        selection: undefined,
-                        messages: [{message: result.messages, time: undefined}],
-                        resultSetSummaries: undefined,
-                        executionElapsed: undefined,
-                        executionEnd: undefined,
-                        executionStart: undefined
-                    }];
-                this.eventEmitter.emit('batchStart', self._batchSets[0]);
-                this.eventEmitter.emit('batchComplete', self._batchSets[0]);
-            } else {
-                // register with the Notification Handler
-                self._notificationHandler.registerRunner(self, queryDetails.ownerUri);
-            }
-            self.eventEmitter.emit('complete');
+            self._notificationHandler.registerRunner(self, queryDetails.ownerUri);
```

Comment:
Shouldn't there be a handler for result.messages since this is how the service returns errors?